tower-sessions-core/src/extract.rs

```diff
@@ -1,20 +1,23 @@
-use async_trait::async_trait;
+use std::future::{ready, Future};
+
 use axum_core::extract::FromRequestParts;
 use http::{request::Parts, StatusCode};
 
 use crate::session::Session;
 
-#[async_trait]
```

Comment:
Removing this and the async_trait import should be the only required changes in this module